easynav_pointcloud_maps_manager/src/easynav_pointcloud_maps_manager/PointCloudMapsManager.cpp

```diff
@@ -0,0 +1,203 @@
+  node->get_parameter(plugin_name + ".map_path_file", map_path_file);
+  node->get_parameter(plugin_name + ".map_topic_in", map_topic_in);
+
+  RCLCPP_INFO(get_node()->get_logger(),
+        "PointCloud :    ..:: PARAMETERS ::..");
+
+  if (package_name != "" && map_path_file != "") {
+    std::string pkgpath;
+    try {
+      pkgpath = ament_index_cpp::get_package_share_directory(package_name);
+      RCLCPP_INFO(get_node()->get_logger(),
+        "PointCloud : Package name = %s", package_name.c_str());
+      map_path_ = pkgpath + "/" + map_path_file;
+      RCLCPP_INFO(get_node()->get_logger(),
+        "PointCloud : Map path file = %s", map_path_.c_str());
+    } catch(ament_index_cpp::PackageNotFoundError & ex) {
+      return std::unexpected("Package " + package_name + " not found. Error: " + ex.what());
+    }
+
+    if (!static_map_.load_from_file(map_path_)) {
+      return std::unexpected("File [" + map_path_ + "] not found");
+    } else {
+      RCLCPP_INFO(get_node()->get_logger(),
+        "PointCloud : Map loaded from file");
+      dynamic_map_.deep_copy(static_map_);
+    }
+  } else {
+    map_path_ = "pointcloud_map.pcd";
+    RCLCPP_INFO(get_node()->get_logger(),
+      "PointCloud : Path Map file to save = ./%s", map_path_.c_str());
+  }
+
+  std::string topic_name;
+  if (map_topic_in != "") {
+    topic_name = map_topic_in;
+    RCLCPP_INFO(get_node()->get_logger(),
+      "PointCloud : Map topic in name = %s", topic_name.c_str());
+  } else {
+    topic_name = node->get_name() + std::string("/") + plugin_name + "/topic_in";
+    RCLCPP_INFO(get_node()->get_logger(),
+        "PointCloud : Input Topic by default");
+  }
+  
+
+  static_map_pub_ = node->create_publisher<sensor_msgs::msg::PointCloud2>(
+    node->get_name() + std::string("/") + plugin_name + "/static_map",
+    rclcpp::QoS(1).transient_local().reliable());
+
+  dynamic_map_pub_ = node->create_publisher<sensor_msgs::msg::PointCloud2>(
+    node->get_name() + std::string("/") + plugin_name + "/dynamic_map", 100);
+
+  incoming_map_sub_ = node->create_subscription<sensor_msgs::msg::PointCloud2>(
+    topic_name, rclcpp::QoS(1).transient_local().reliable(),
+    [this](sensor_msgs::msg::PointCloud2::UniquePtr msg) {
+
+      RCLCPP_INFO(get_node()->get_logger(),
+      "PointCloud : topic_callback : Reading Map");
+      static_map_.from_point_cloud(*msg);
+
+      static_map_.to_point_cloud(static_map_msg_);
+      static_map_msg_.header.frame_id = "map";
+      static_map_msg_.header.stamp = this->get_node()->now();
+
+      static_map_pub_->publish(static_map_msg_);
+    });
+
+  savemap_srv_ = node->create_service<std_srvs::srv::Trigger>(
+    node->get_name() + std::string("/") + plugin_name + "/savemap",
+    [this](
+      const std::shared_ptr<std_srvs::srv::Trigger::Request> request,
+      std::shared_ptr<std_srvs::srv::Trigger::Response> response)
+    {
+      (void)request;
+
+      if (!static_map_.save_to_file(map_path_)) {
```

Comment:
Initialize `map_path_` to before line 69 to a default value, like `/tmp/pc.pcd`. So a map can be saved even if no package/filename has been set. This is useful when mapping.